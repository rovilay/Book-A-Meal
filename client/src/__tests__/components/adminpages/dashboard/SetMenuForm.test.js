import React from 'react';
import { shallow } from 'enzyme';
import SetMenuForm from '../../../../components/adminpages/dashboard/MenuCard/SetMenuForm';
import { menuMeals } from '../../../__mockData__/menuMock';

const props = {
  submitNewMenu: jest.fn(),
  newMenuMeals: menuMeals,
  meals: [],
  mealPagination: {},
  setNewMenuMeal: jest.fn()
};

describe('SetMenuForm component test', () => {
  const setup = () => {
    return shallow(<SetMenuForm {...props} />)
  };


  it('should render SetMenuForm correctly', (done) => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();

    done();
  });

  it('should call `submitNewMenu` on create menu form submition', (done) => {
    const wrapper = setup();

    const event = {
      preventDefault: jest.fn()
    }
    wrapper.find(`.setmenu-form`).first().simulate('submit', event);
    expect(props.submitNewMenu).toHaveBeenCalled()

    done();
  });
});
