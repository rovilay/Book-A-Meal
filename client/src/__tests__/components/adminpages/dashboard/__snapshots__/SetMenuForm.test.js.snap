// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SetMenuForm component test should render SetMenuForm correctly 1`] = `
<form
  className="setmenu-form"
  onSubmit={[Function]}
>
  <div
    className="date"
  >
    <label
      htmlFor="date"
    >
      Post on:
    </label>
    <p
      className="input-div"
    >
      <input
        id="postOn"
        max="2018-09-12"
        min="2018-09-10"
        name="menu-date"
        placeholder="DD / MM / YYYY"
        required={true}
        type="date"
      />
    </p>
  </div>
  <hr />
  <MealCheckBoxCard
    mealPagination={Object {}}
    meals={Array []}
    newMenuMeals={
      Array [
        Object {
          "id": "1",
          "title": "EBA AND EGUSI",
        },
        Object {
          "id": "2",
          "title": "SALAD",
        },
        Object {
          "id": "3",
          "title": "SPAGHETTI",
        },
        Object {
          "id": "4",
          "title": "YAM AND EGG",
        },
      ]
    }
    setNewMenuMeal={[MockFunction]}
    submitNewMenu={[MockFunction]}
  />
  <hr />
  <p
    className="submit-btn"
  >
    <button
      className="btn-2"
      disabled={false}
      id="addbtn"
      name="addbtn"
      type="submit"
    >
      Add Menu
    </button>
  </p>
</form>
`;
